geopolars/src/geoseries.rs

```diff
@@ -503,35 +551,46 @@ impl GeoSeries for Series {
         Series::try_from(("result", Arc::new(result) as ArrayRef))
     }
 
-    fn rotate(&self, angle: f64, origin: TransformOrigin) -> Result<Series> {
+    fn rotate<'a>(
+        &self,
+        angle: impl Into<BroadcastableParameter<'a, f64>>,
+        origin: TransformOrigin,
+    ) -> Result<Series> {
         use geo::algorithm::bounding_rect::BoundingRect;
         use geo::algorithm::centroid::Centroid;
-        match origin {
```

Comment:
I feel like this could get messy quickly, where we have three different enum values for `TransformOrigin` and two different enum values for `BroadcastableParameter`, though I don't immediately have a better idea